src/Courses/TopicGrades/GradeInfoHeader.tsx

```diff
@@ -42,154 +41,260 @@ const useStyles = makeStyles((theme) => ({
 }));
 
 export const GradeInfoHeader: React.FC<GradeInfoHeaderProps> = ({
-    grade,
-    workbookId,
     selected,
     setSelected,
-    onSuccess,
-    topicId
+    topic
 }) => {
-    const displayCurrentScore = useRef<string | null>(null);
-    if (_.isNil(displayCurrentScore.current)) {
-        displayCurrentScore.current = (grade.effectiveScore * 100).toFixed(1);
-    }
     const classes = useStyles();
     const [showGradeModal, setShowGradeModal] = useState<boolean>(false);
     const [topicGrade, setTopicGrade] = useState<number | null>(null);
+    const [grade, setGrade] = useState<StudentGrade | null>(null);
     const [info, setInfo] = useState<{
-        legalScore: number;
-        overallBestScore: number;
-        partialCreditBestScore: number;
-        effectiveScore: number;
+        legalScore?: number;
+        overallBestScore?: number;
+        partialCreditBestScore?: number;
+        effectiveScore?: number;
         workbookId?: number;
+        studentGradeId?: number;
+        studentGradeInstanceId?: number;
         averageScore?: number;
         attemptsCount?: number;
         versionMap?: Record<number, Array<number>>;
-        workbookList?: WorkbookOption[];
-    }>({
-        legalScore: grade.legalScore,
-        overallBestScore: grade.overallBestScore,
-        partialCreditBestScore: grade.partialCreditBestScore,
-        effectiveScore: grade.effectiveScore,
-        workbookId,
-    });
+        // workbookList?: WorkbookOption[];
+    }>({});
+    const displayCurrentScore = useRef<string | null>(null);
```

Comment:
Looks like this was already there, but why do we need to useRef here?

Reply:
IDK, I thought you added this for showing the topic total score

Reply:
I did because we needed to manage the parameter that came in vs the current state (show that you have a 75 and then an input that lets you make it 100)

But we are reusing that component right? or did we copy it?